### ISSUE TITLE
torch int times python float is int

### ISSUE THREAD

**Issue by petered** [pytorch/pytorch#15056]
## 🐛 Bug

When multiplying an integer tensor by a float, pytorch does not cast the result to float.
```
>>> torch.tensor(1)*.5
tensor(0)
```
This can easily result in unexpected outcomes, and seems to be almost always an undesirable behaviour.  Compare to numpy's behaviour:

```
>>> np.array(1)*.5
0.5
```

 - PyTorch Version (e.g., 1.0): 1.0
 - OS (e.g., Linux): MacOS
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Python version: 3.6
 - CUDA/cuDNN version: None

**Comment by petered** [comment]
This just came back to bite me again, it'd be really great to fix this!

**Comment by kaczorrrro** [comment]
Just hit the same issue when trying to select biggest numbers in my tensor
```
# values : int Tensor
T.where(values > (max(values) * 0.9))
```

**Comment by mruberry** [comment]
This is fixed in PyTorch 1.7 (and possibly earlier versions of PyTorch, too).
